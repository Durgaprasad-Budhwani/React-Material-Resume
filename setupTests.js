/**
 * Created by Durgaprasad Budhwani on 12/14/2017.
 */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Element.prototype.getBoundingClientRect = function(){
    return {
        y: -100,
        top: -100
    }
};

configure({ adapter: new Adapter() });