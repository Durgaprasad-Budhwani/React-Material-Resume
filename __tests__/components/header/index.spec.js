/**
 * Created by Durgaprasad Budhwani on 10/9/2017.
 */
/* global expect,it, describe */
import React, {Component} from 'react';
import HeaderContainer, {Header}  from '../../../src/components/header';
import renderer from 'react-test-renderer';
import {createMount} from '@material-ui/core/test-utils';
import toJson from 'enzyme-to-json';
import '../../../setupTests';

describe('<Header />', ()=> {
    const classes = {
        header: {
        
        },
        headerBar: {
        
        },
        headerBarFixed: {
        
        },
        headerBackground: {
        
        },
        flex: {
        },
        horizontalNavigation: {
        
        },
        verticalNavigation: {
        
        },
        hamburger: {
        
        }
    };
    
    let mount;
    beforeEach(() => {
        mount = createMount()
    });
    
    afterEach(() => {
        mount.cleanUp();
    });
    
    it('renders header container', () => {
        const tree = renderer.create(<HeaderContainer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders header ', () => {
        const tree = renderer.create(<Header classes={classes}/>);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    
    it('renders header with drawer is hidden', () => {
        const wrapper = mount(<Header classes={classes}/>);
        const instance = wrapper.instance();
        instance._hideDrawer();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
    it('renders header with handle scroll event', () => {
        const wrapper = mount(<Header classes={classes}/>);
        const instance = wrapper.instance();
       
        instance.handleScroll();
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

