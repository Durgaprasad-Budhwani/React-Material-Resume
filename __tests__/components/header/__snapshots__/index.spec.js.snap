// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<div
  className="Header-header-1"
>
  <div
    className="Header-headerBackground-3"
  />
  <header
    className="MuiPaper-root-14 MuiPaper-shadow4-20 MuiAppBar-root-7 MuiAppBar-positionFixed-8 MuiAppBar-colorPrimary-12 mui-fixed Header-headerBar-2"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-gutters-42"
    >
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-52 MuiIconButton-root-43 MuiIconButton-colorContrast-45"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label-49"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-55"
        />
      </button>
      <div
        className="Header-flex-4"
      />
      <div>
        Profile 
      </div>
      <div>
        Contact 
      </div>
    </div>
  </header>
</div>
`;
